include/cpp-statsd-client/UDPSender.hpp

```diff
@@ -161,27 +163,33 @@ inline UDPSender::~UDPSender() {
         return;
     }
 
-    if (m_batching) {
+    // If we're running a background thread tell it to stop
+    if (m_batchingThread.joinable()) {
```

Comment:
`m_batching` really just meant are we running a background thread, which is easily checkable with the `joinable` function